sklearn/datasets/_openml.py

```diff
@@ -478,59 +512,110 @@ def _download_data_to_bunch(
 
-    col_slice_y = [int(features_dict[col_name]["index"]) for col_name in target_columns]
+    sparse : bool
+        Whether or not the dataset will be a sparse ARFF format.
 
-    col_slice_x = [int(features_dict[col_name]["index"]) for col_name in data_columns]
-    for col_idx in col_slice_y:
-        feat = features_list[col_idx]
-        nr_missing = int(feat["number_of_missing_values"])
-        if nr_missing > 0:
-            raise ValueError(
-                "Target column {} has {} missing values. "
-                "Missing values are not supported for target "
-                "columns. ".format(feat["name"], nr_missing)
-            )
+    data_home : str
+        The location where to cache the data.
 
-    # Access an ARFF file on the OpenML server. Documentation:
-    # https://www.openml.org/api_data_docs#!/data/get_download_id
+    as_frame : bool
+        Whether or not to return the data into a pandas DataFrame.
 
-    if as_frame:
-        output_arrays_type = "pandas"
-    elif sparse:
+    columns_info_openml_as_list : list of dict
+        The information regarding the columns provided by OpenML for the
+        ARFF dataset. The information is stored as a list of dictionaries.
+
+    data_columns : list of str
+        The list of the features to be selected.
+
+    target_columns : list of str
+        The list of the target variables to be selected.
+
+    shape : tuple or None
+        With `parser="liac-arff"`, when using a generator to load the data,
+        one needs to provide the shape of the data beforehand.
+
+    md5_checksum : str
+        The MD5 checksum provided by OpenML to check the data integrity.
+
+    n_retries : int, default=3
+        Number of retries when HTTP errors are encountered. Error with status
+        code 412 won't be retried as they represent OpenML generic errors.
+
+    delay : float, default=1.0
+        Number of seconds between retries.
+
+    parser : {"liac-arff", "pandas"}
+        The parser used to parse the ARFF file.
+
+    Returns
+    -------
+    data : :class:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        X : {ndarray, sparse matrix, dataframe}
+            The data matrix.
+        y : {ndarray, dataframe, series}
+            The target.
+        frame : dataframe or None
+            A dataframe containing both `X` and `y`. `None` if
+            `output_array_type != "pandas"`.
+        nominal_attributes : list of str or None
+            The names of the features that are categorical. `None` if
+            `output_array_type == "pandas"`.
+    """
+    # Prepare which columns and data types should be returned for the X and y
+    features_dict = {
+        feature["name"]: feature for feature in columns_info_openml_as_list
+    }
+
+    if sparse:
         output_arrays_type = "sparse"
+    elif as_frame:
+        output_arrays_type = "pandas"
     else:
         output_arrays_type = "numpy"
```

Comment:
Shouldn't we raise a `ValueError` when `as_frame and sparse` at the same time?

Reply:
It is early caught in `fetch_openml`